src/devnet-utils.ts

```diff
@@ -117,6 +118,34 @@ Make sure you really want to interact with Devnet and that it is running and ava
         return response.data;
     }
 
+    public async estimateMessageFee(
+        fromAddress: string,
+        toAddress: string,
+        funcionName: string,
+        payload: number[]
+    ) {
+        const body = {
+            from_address: hexToDecimalString(fromAddress),
+            to_address: toAddress,
+            entry_point_selector: hash.getSelectorFromName(funcionName),
+            payload: payload.map((item) => numericToHexString(item))
+        };
+
+        const response = await this.requestHandler<FeeEstimation>(
+            "/feeder_gateway/estimate_message_fee",
```

Comment:
If this doesn't interact with a specific devnet endpoint, I don't think it should be inside this Devnet wrapper class. I.e. it interacts with a feeder_gateway endpoint.

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/312/commits/ee4a0effa3fe3b460a6e0be8a167ca641f19a5a7